modules/metastorage/src/integrationTest/java/org/apache/ignite/internal/metastorage/impl/ItMetaStorageMultipleNodesAbstractTest.java

```diff
@@ -585,6 +588,66 @@ void testIdleSafeTimePropagationLeaderTransferred(TestInfo testInfo) throws Exce
         assertThat(secondNodeTime.waitFor(now), willCompleteSuccessfully());
     }
 
+    /**
+     * Tests that idle safe time propagation does not advance safe time while watches of a normal command are being executed.
+     */
+    @Test
+    void testIdleSafeTimePropagationAndNormalSafeTimePropagationInteraction(TestInfo testInfo) throws Exception {
+        // Enable idle safe time sync.
+        CompletableFuture<Void> updateIdleSyncTimeIntervalFuture = metaStorageConfiguration.idleSyncTimeInterval().update(100L);
+        assertThat(updateIdleSyncTimeIntervalFuture, willCompleteSuccessfully());
+
+        Node firstNode = startNode(testInfo);
```

Comment:
Do we really need a multi-node test here? I can imagine a "unit" test where we start a MetaStorageManager, block its watches and then manually run a time sync command.